lib/p5.play.js

```diff
@@ -4743,24 +4743,45 @@ function SpriteSheet(pInst) {
 
   if(spriteSheetArgs[0] instanceof p5.Image || spriteSheetArgs[0] instanceof Image) {
     this.image = spriteSheetArgs[0];
-    if (spriteSheetArgs.length === 4) {
+    if (longArgs) {
       this._generateSheetFrames();
     }
   } else {
     // When the final argument is present (either the 3rd or the 5th), it indicates
     // whether we should load the URL as an Image element (as opposed to the default
-    // behavior, which is to load it as a p5.Image)
+    // behavior, which is to load it as a p5.Image). If that argument is a function,
+    // it will be called back once the load succeeds or fails. On success, the Image
+    // will be supplied as the only parameter. On failure, null will be supplied.
+    var callback;
     if (shortArgs) {
       if (spriteSheetArgs[2]) {
-        this.image = pInst.loadImageElement(spriteSheetArgs[0]);
+        if (typeof spriteSheetArgs[2] === 'function') {
+          callback = spriteSheetArgs[2];
+        }
+        this.image = pInst.loadImageElement(
+          spriteSheetArgs[0],
+          function(img) { if (callback) return callback(img); },
+          function() { if (callback) return callback(null); }
+        );
       } else {
         this.image = pInst.loadImage(spriteSheetArgs[0]);
       }
     } else if (longArgs) {
+      var generateSheetFrames = this._generateSheetFrames.bind(this);
       if (spriteSheetArgs[4]) {
-        this.image = pInst.loadImageElement(spriteSheetArgs[0], this._generateSheetFrames.bind(this));
+        if (typeof spriteSheetArgs[4] === 'function') {
+          callback = spriteSheetArgs[4];
+        }
```

Comment:
No need right now but I'd love for us to break the argument decomposition out of the other logic in this function.